src/Fixer/Whitespace/ReferenceSpacesFixer.php

```diff
@@ -0,0 +1,268 @@
+ */
+final class ReferenceSpacesFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /**
+     * @internal
+     */
+    public const TYPE_ASSIGNMENT = 'assignment';
+
+    /**
+     * @internal
+     */
+    public const TYPE_FUNCTION_SIGNATURE = 'function_signature';
+
+    /**
+     * @internal
+     */
+    public const TYPE_ANONYMOUS_FUNCTION_USE_BLOCK = 'anonymous_function_use_block';
+
+    /**
+     * @internal
+     */
+    public const BY_ASSIGN = 'by_assign';
+
+    /**
+     * @internal
+     */
+    public const BY_REFERENCE = 'by_reference';
+
+    /**
+     * @internal
+     */
+    public const SINGLE_SPACE = 'single_space';
+
+    /**
+     * @internal
+     */
+    public const NO_SPACE = 'no_space';
+
+    /** @var array<'&'|287|288> */
+    private array $relevantTokens = [];
+
+    public function configure(array $configuration): void
+    {
+        parent::configure($configuration);
+
+        $this->relevantTokens = \PHP_VERSION_ID >= 80100
+            ? [T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG, T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG, '&']
+            : ['&'];
+    }
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Reference operator should be surrounded by space as defined.',
+            [
+                new CodeSample(
+                    '<?php $foo = & $bar;
+'
+                ),
+                new CodeSample(
+                    '<?php $foo = & $bar;
+',
+                    ['assignment' => self::BY_ASSIGN]
+                ),
+                new CodeSample(
+                    '<?php $foo = & $bar;
+',
+                    ['assignment' => self::BY_REFERENCE]
+                ),
+                new CodeSample(
+                    '<?php $foo =&$bar;
+',
+                    ['assignment' => self::SINGLE_SPACE]
+                ),
+                new CodeSample(
+                    '<?php $foo =& $bar;
+',
+                    ['assignment' => self::NO_SPACE]
+                ),
+                new CodeSample(
+                    '<?php function foo(& $bar) {}
+',
+                    ['function_signature' => self::BY_REFERENCE]
+                ),
+                new CodeSample(
+                    '<?php function foo(&$bar) {}
+',
+                    ['function_signature' => self::SINGLE_SPACE]
+                ),
+                new CodeSample(
+                    '<?php $foo = function () use (& $bar) {};
+',
+                    ['anonymous_function_use_block' => self::BY_REFERENCE]
+                ),
+                new CodeSample(
+                    '<?php $foo = function () use (&$bar) {};
+',
+                    ['anonymous_function_use_block' => self::SINGLE_SPACE]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run after BinaryOperatorSpacesFixer
+     */
+    public function getPriority(): int
+    {
+        return -33;
```

Comment:
the failing test that you see is because you put a non-zero priority here. it is asking you to create an explicit priority testing case in `tests/Fixtures/Integration/priority/` , and entry in `getFixersPriorityGraph` you mentioned